nixos/modules/services/infrastructure/terraform-cloud-agent/default.nix

```
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.terraform-cloud-agent;
+in
+{
+  options.services.terraform-cloud-agent = {
+    enable = lib.mkEnableOption "terraform-cloud-agent";
+
+    tokenPath = lib.mkOption {
+      type = lib.types.path;
+      default = "%E/token";
+    };
+
+    data-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    cache-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    flags = lib.mkOption {
+      freeformType = lib.types.attrsOf lib.types.str;
+
+      name = lib.mkOption {
+        type = lib.types.str;
+      };
+
+      log-level = lib.mkOption {
+        type = lib.types.nullOr (lib.types.enum [ "trace" "debug" "info" "warn" "error" ]);
+        default = null;
+      };
+
+      address = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+      };
+
+      accept = lib.mkOption {
+        type = lib.types.commas;
+        default = "plan,apply,policy,assessment";
+      };
+
+      auto-update = lib.mkOption {
+        type = lib.types.str;
+        readOnly = true;
+        default = "disabled";
+      };
+    };
+  };
+
+  config.systemd.services.terraform-cloud-agent = lib.mkIf cfg.enable {
+    description = "Agent that executes plans from Terraform Cloud";
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network.target" ];
```

Is this the target we want?
```suggestion
    after = [ "network-online.target" ];
```

---

Uh yeah, I think you're right.

---

Addressed in ba8b81567043.